Spec/RealtimeClient.swift

```diff
@@ -352,6 +352,478 @@ class RealtimeClient: QuickSpec {
+
+                    waitUntil(timeout: testTimeout) { done in
+                        client.connection.once(.Connecting) { stateChange in
+                            expect(stateChange?.reason).to(beNil())
+
+                            let authOptions = ARTAuthOptions()
+                            authOptions.key = AblyTests.commonAppSetup().key
+
+                            client.auth.authorize(nil, options: authOptions) { tokenDetails, error in
+                                expect(error).to(beNil())
+                                guard let tokenDetails = tokenDetails else {
+                                    fail("TokenDetails is nil"); done(); return
+                                }
+                                expect(tokenDetails.token).toNot(beNil())
+                                expect(client.connection.state).to(equal(ARTRealtimeConnectionState.Connected))
+
+                                guard let transport = client.transport as? TestProxyTransport else {
+                                    fail("TestProxyTransport is not set"); done(); return
+                                }
+                                expect(transport.protocolMessagesReceived.filter({ $0.action == .Connected })).to(haveCount(1))
+                                done()
+                            }
+                        }
+                        client.connect()
+                    }
+
+                    expect(client.connection.state).toEventually(equal(ARTRealtimeConnectionState.Connected), timeout: testTimeout)
+                }
+
+                // RTC8b1
+                it("authorize call should complete with the new token once the connection has moved to the CONNECTED state, or with an error if the connection instead moves to the FAILED, SUSPENDED, or CLOSED states") {
+                    let options = AblyTests.clientOptions()
+                    options.autoConnect = false
+                    let testToken = getTestToken()
+                    options.token = testToken
+                    let client = ARTRealtime(options: options)
+                    defer { client.dispose(); client.close() }
+                    client.setTransportClass(TestProxyTransport.self)
+
+                    waitUntil(timeout: testTimeout) { done in
+                        client.connection.once(.Connecting) { stateChange in
+                            expect(stateChange?.reason).to(beNil())
+
+                            let tokenParams = ARTTokenParams()
+                            tokenParams.clientId = "john"
+
+                            let authOptions = ARTAuthOptions()
+                            authOptions.authCallback = { tokenParams, completion in
+                                completion(getTestTokenDetails(clientId: "tester"), nil)
+                            }
+
+                            client.auth.authorize(tokenParams, options: authOptions) { tokenDetails, error in
```

Comment:
I am not sure you have tested this correctly, the connection is becoming connected anyway given you have called connect.  Why not instead call `authorize` on the connection which is not configured to autoConnect, and simply check that the connection is `CONNECTED` once the callback is called?  Looking below, it seems you are checking that the state is `CONNECTING` which is wrong according to the spec "authorize call should complete with the new token once the connection has moved to the CONNECTED".

Also, you are not testing that it's called only once moved to the FAILED state, can you test that or do you feel it's covered sufficiently elsewhere?  If so, perhaps add a comment to that test.